tests/admin_views/tests.py

```diff
@@ -7299,6 +7299,24 @@ def test_password_mismatch(self):
             ["The two password fields didn’t match."],
         )
 
+    def test_guideline_username(self):
+        username = User._meta.get_field(User.USERNAME_FIELD).verbose_name
+        response = self.client.get(reverse("admin:auth_user_add"))
+        self.assertContains(
+            response,
+            "<p>First, fill the fields: %s and password. "
+            "Then, you’ll be able to edit more user options.</p>" % (username),
+            html=True,
+        )
+        response = self.client.get(
+            reverse("admin:auth_user_add") + "?%s=1" % IS_POPUP_VAR
+        )
+        self.assertContains(
+            response,
+            "<p>Fill the fields: %s and password.</p>" % (username),
+            html=True,
+        )
```

Comment:
I think we want to test an admin with a custom user, something like...
```diff
--- a/tests/auth_tests/test_views.py
+++ b/tests/auth_tests/test_views.py
@@ -8,6 +8,7 @@ from urllib.parse import quote, urljoin
 from django.apps import apps
 from django.conf import settings
 from django.contrib.admin.models import LogEntry
+from django.contrib.admin.options import IS_POPUP_VAR
 from django.contrib.auth import BACKEND_SESSION_KEY, REDIRECT_FIELD_NAME, SESSION_KEY
 from django.contrib.auth.forms import (
     AuthenticationForm,
@@ -1758,3 +1759,40 @@ class UUIDUserTests(TestCase):
         # constraint error, delete the entry before constraints are checked
         # after the test.
         row.delete()
+
+
+@override_settings(
+    AUTH_USER_MODEL="auth_tests.CustomUser",
+    ROOT_URLCONF="auth_tests.urls_custom_user_admin",
+)
+class CustomUserAdminTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.superuser = CustomUser._default_manager.create_superuser(
+            email="test@example.com",
+            password="test",
+            date_of_birth=datetime.date(2006, 4, 25),
+        )
+
+    def setUp(self):
+        self.client.force_login(self.superuser)
+
+    def test_guideline_username(self):
+        response = self.client.get(
+            reverse("custom_user_admin:auth_tests_customuser_add")
+        )
+        self.assertContains(
+            response,
+            "<p>First, fill the email address and password fields. "
+            "Then, you’ll be able to edit more user options.</p>",
+            html=True,
+        )
+        response = self.client.get(
+            reverse("custom_user_admin:auth_tests_customuser_add")
+            + "?%s" % IS_POPUP_VAR
+        )
+        self.assertContains(
+            response,
+            "<p>Fill the email address and password fields.</p>",
+            html=True,
+        )
diff --git a/tests/auth_tests/urls_custom_user_admin.py b/tests/auth_tests/urls_custom_user_admin.py
index 1c7ce1eb42..33f051c467 100644
--- a/tests/auth_tests/urls_custom_user_admin.py
+++ b/tests/auth_tests/urls_custom_user_admin.py
@@ -7,6 +7,14 @@ site = admin.AdminSite(name="custom_user_admin")
 
 
 class CustomUserAdmin(UserAdmin):
+    fieldsets = ((None, {"fields": (get_user_model().USERNAME_FIELD, "password")}),)
+    add_fieldsets = (
+        (
+            None,
+            {"fields": (get_user_model().USERNAME_FIELD, "password1", "password2")},
+        ),
+    )
+
     def log_change(self, request, obj, message):
         # LogEntry.user column doesn't get altered to expect a UUID, so set an
```

Reply:
Reflecting the test code you suggested, there is no problem with the test in my local environment, but errors occur in the test process by PR.

**Reverse for 'auth_tests_uuiduser_change' not found. 'auth_tests_uuiduser_change' is not a valid view function or pattern name.**

```
Traceback (most recent call last):
  File "/home/jenkins/workspace/pull-requests-focal/database/mysql/label/focal-pr/python/python3.12/tests/auth_tests/test_views.py", line 1729, in test_admin_password_change
    user_change_url = reverse(
                      ^^^^^^^^
```

As above, there is a problem in finding the logic associated with url through the reverse function and path_name.

I don't know why, but I'm having a problem with a path that uses the same ROOT_URLCONF value, urls_custom_user_admin.

Do you know anything about how to solve this problem?
I will also look for more to resolve this error. 💪

Reply:
Ah sorry, I pushed up a tweak. I think the test ordering and initialization of `get_user_model()` was causing a problem

Reply:
@sarahboyce Thank you for your help. 💪💪